### ISSUE TITLE
`#![cfg(platform)]` causes clippy to ignore `#![allow(...)]` directives

### ISSUE THREAD

**Issue by briansmith** [rust-lang/rust-clippy#6185]
I tried this code:

```rust
#![cfg(windows)]
#![allow(missing_docs)]

pub fn a_function() {}
```

I am invoking clippy as `cargo clippy -- --deny missing_docs`

I expected to see this happen:

On all targets/hosts, I expect clippy to report no problems.

Instead, this happened:

On Windows, clippy did the right thing.
On Linux, I got:

```
error: missing documentation for crate
 --> src/lib.rs:1:1
  |
1 | / #![cfg(windows)]
2 | | #![allow(missing_docs)]
3 | |
4 | | pub fn a_function() {}
  | |______________________^
 ```

### Meta

clippy 0.0.212 (204bb9b 2020-03-17)
rustc 1.43.0 (4fb7144ed 2020-04-20)
binary: rustc
commit-hash: 4fb7144ed159f94491249e86d5bbd033b5d60550
commit-date: 2020-04-20
host: x86_64-unknown-linux-gnu
release: 1.43.0
LLVM version: 9.0

**Comment by ebroto** [comment]
FYI `missing_docs` is a rustc lint, not a Clippy lint.

`cargo check` yields the same error in Linux:
```terminal
❯ RUSTFLAGS='--deny missing_docs' cargo check
    Checking pepe v0.1.0 (/home/ebroto/tst/pepe)
error: missing documentation for the crate
 --> src/lib.rs:1:1
  |
1 | / #![cfg(windows)]
2 | | #![allow(missing_docs)]
3 | |
4 | | pub fn a_function() {}
  | |______________________^
  |
  = note: requested on the command line with `-D missing-docs`

error: aborting due to previous error

error: could not compile `pepe`

To learn more, run the command again with --verbose.
```

IIUC the whole crate should be omitted from compilation, so I'm not sure why it sees the function in the first place :thinking: 

Maybe it's worth it to file an issue in rust-lang/rust?
